test/widgets/action_sheet_test.dart

```diff
@@ -551,6 +554,93 @@ void main() {
       final expectedLink = narrowLink(store, narrow, nearMessageId: message.id).toString();
       check(await Clipboard.getData('text/plain')).isNotNull().text.equals(expectedLink);
     });
+
+    testWidgets('Full Url Testing', (tester) async {
+      final message = eg.streamMessage();
+      final narrow = TopicNarrow.ofMessage(message);
+
+      final customAccount = eg.selfAccount.copyWith(
+          id: 2, realmUrl: Uri.parse('https://chat.example'));
+      await setupToMessageActionSheet(
+          tester, message: message, narrow: narrow, account: customAccount);
+      final customAccountStore = await testBinding.globalStore.perAccount(customAccount.id);
+
+      await tapCopyMessageLinkButton(tester);
+      await tester.pump(Duration.zero);
+      final expectedLink = narrowLink(
+          customAccountStore, narrow, nearMessageId: message.id).toString();
+      check(await Clipboard.getData('text/plain')).isNotNull().text.equals(expectedLink);
+    });
+  });
+
+  group('Message link verbatim URL tests', () {
+
+    Future<void> tapCopyMessageLinkButton(WidgetTester tester) async {
+      await tester.ensureVisible(find.byIcon(Icons.link, skipOffstage: false));
+      await tester.tap(find.byIcon(Icons.link));
+      await tester.pump(); // [MenuItemButton.onPressed] called in a post-frame callback: flutter/flutter@e4a39fa2e
+    }
+
+    testWidgets('Copies message link with verbatim, full URL output', (tester) async {
+      final stream = eg.stream(streamId: 123, name: 'stream 123');
+      final message = eg.streamMessage(
+          id: 13579, stream: stream, topic: 'example topic 341');
+      final narrow = TopicNarrow.ofMessage(message);
+      final account = eg.selfAccount.copyWith(
+          realmUrl: Uri.parse('https://chat.zulip.org'));
+      await setupToMessageActionSheet(
+          tester, message: message, narrow: narrow, account: account);
+
+      final store = await testBinding.globalStore.perAccount(account.id);
+      if (!store.streams.containsKey(stream.streamId)) {
+        await store.addStream(stream);
+      }
+
+      await tapCopyMessageLinkButton(tester);
+      await tester.pump(Duration.zero);
+      const expectedLink = 'https://chat.zulip.org/#narrow/stream/123-stream-123/topic/example.20topic.20341/near/13579';
+      check(await Clipboard.getData('text/plain')).isNotNull().text.equals(expectedLink);
+    });
+
+    testWidgets('Copies link with special characters in stream name and topic', (tester) async {
+      final stream = eg.stream(streamId: 42, name: 'weird & wonderful');
+      final message = eg.streamMessage(
+          id: 98765, stream: stream, topic: 'Whats up? 123!');
+      final narrow = TopicNarrow.ofMessage(message);
+      final account = eg.selfAccount.copyWith(
+          realmUrl: Uri.parse('https://chat.zulip.org'));
```

Comment:
It should be fine to reuse the same set of stream and topic, because we are more interested in the variants of `realmUrl`. In addition to this one, let's also try something like `https://chat.zulip.org/path` and `https://chat.zulip.org/`. Since most of these tests will be similar, extracting a helper will be simplify the code.